wandb/sdk/launch/agent2/controllers/base.py

```diff
@@ -157,3 +193,146 @@ async def find_orphaned_jobs(self) -> List[Any]:
+
+        if not run_called_init:
+            fnames = None
+            if logs:
+                fnames = tracker.saver.save_contents(logs, "error.log", "error")
+            if status == "finished":
+                _msg = "The submitted job exited successfully but failed to call wandb.init"
+            else:
+                _msg = f"The submitted job failed to call wandb.init exited with status: {status}"
+
+            fail_run_queue_item = event_loop_thread_exec(
+                self.jobset.api.fail_run_queue_item
+            )
+            await fail_run_queue_item(
+                item_id,
+                _msg,
+                "run",
+                fnames,
+            )
+
+    async def fail_run_with_exception(
+        self,
+        item_id: str,
+        exception: Exception,
+        tracker: Optional[JobAndRunStatusTracker] = None,
+    ) -> None:
+        tb_str = traceback.format_exception(
+            type(exception), value=exception, tb=exception.__traceback__
+        )
+        fnames = None
+        phase = None
+        if tracker is not None:
+            phase = tracker.err_stage
+            fnames = tracker.saver.save_contents("".join(tb_str), "error.log", "error")
+        else:
+            phase = "agent"
+
+        fail_run_queue_item = event_loop_thread_exec(
+            self.jobset.api.fail_run_queue_item
+        )
+        await fail_run_queue_item(item_id, str(exception), phase, fnames)
+
+        return
+
+    async def fail_unsubmitted_run(self, item_id: str) -> None:
+        fail_run_queue_item = event_loop_thread_exec(
+            self.jobset.api.fail_run_queue_item
+        )
+        await fail_run_queue_item(
+            item_id,
+            "The job was not submitted successfully",
+            "agent",
+        )
+        return
+
+
+async def check_run_called_init(
+    api: Api,
+    run: AbstractRun,
+    entity: str,
+    project: str,
+    run_id: str,
+    run_queue_item_id: str,
+) -> Tuple[bool, Optional[str]]:
+    called_init = False
+    # We do some weird stuff here getting run info to check for a
+    # created in run in W&B.
+    #
+    # We retry for 60 seconds with an exponential backoff in case
+    # upsert run is taking a while.
+    logs = None
+    start_time = time.time()
+    interval = 1
+    while True:
+        called_init = await check_run_exists_and_inited(
+            api,
+            entity,
+            project,
+            run_id,
+            run_queue_item_id,
+        )
+        if called_init or time.time() - start_time > RUN_INFO_GRACE_PERIOD:
+            break
+        if not called_init:
+            # Fetch the logs now if we don't get run info on the
+            # first try, in case the logs are cleaned from the runner
+            # environment (e.g. k8s) during the run info grace period.
+            if interval == 1:
+                logs = await run.get_logs()
+            await asyncio.sleep(interval)
+            interval *= 2
```

Comment:
IMO we should have an explicit cap on this interval

Reply:
We can do that, right now this cap is controlled via RUN_INFO_GRACE_PERIOD where if the waited interval is > RUN_INFO_GRACE_PERIOD (60s) we just break